### PR TITLE
Model Vulnerability Metrics + Inception-Only Config

### PR DESCRIPTION
**Pull request by philswatton**
This PR:

- Replaces the old measure of model vulnerability with the same measures used for transfer success
- Updates both the attack script and src/ files to achieve this
- Adds the inception-only config with MMD, OTDD, and PAD. Does not include KDE as these required a small number of features to work well
- Some fixes/corrections

I have run another of the previously failing attacks and can confirm everything is working as expected on Baskerville

### CODE REVIEW

File: src/modsim2/attack/adversarial_images.py [hunk 1]
```diff
@@ -148,6 +154,9 @@ def generate_adversarial_images(
                         L2FastGradientAttack or BoundaryAttack
         epsilons: Pertubation parameter for the attacks
         device: String passed to fb.PyTorchModel for computation
+        batch_size: int,
+        trainer_kwargs: dict = {},
+        loss_function: Callable = torch.nn.functional.nll_loss,
         **kwargs: Additional arguments based to attack setup
 
     Returns: a torch.tensor containing the adversarial images and a torch.tensor
```
Comment (joannacknight) [inline src/modsim2/attack/adversarial_images.py:162]: I think this statement needs a small update to describe the vulnerability metric?

**Review by joannacknight — CHANGES_REQUESTED** [review]
Just some minor comments

File: configs/metrics/inception.yaml [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+      max_samples: 10000
+  pad_linear_inception:
+    class: pad
+    arguments:
+      kernel_name: linear
+      c_values:
+        - 0.1
+        - 1
+        - 10
+      embedding_name: inception_umap
```
[the remark this answers is not in the document]
Comment (philswatton) [reply configs/metrics/inception.yaml:75]: Yes, good spot!

File: src/modsim2/attack/adversarial_images.py (hunk 1 above)
Comment (philswatton) [reply src/modsim2/attack/adversarial_images.py:162]: Have updated

**Review by joannacknight — CHANGES_REQUESTED** [review]
One small typo and a suggestion about the name of a function.

File: src/modsim2/attack/adversarial_images.py [hunk 3]
```diff
@@ -148,11 +154,15 @@ def generate_adversarial_images(
...
         device: String passed to fb.PyTorchModel for computation
+        batch_size: int,
+        trainer_kwargs: dict = {},
+        loss_function: Callable = torch.nn.functional.nll_loss,
         **kwargs: Additional arguments based to attack setup
 
-    Returns: a torch.tensor containing the adversarial images and a torch.tensor
-             where each element represents the percentage of successful attacks at
-             each value of epsilon
+    Returns: a torch.tensor containing the adversarial images and a ductionary
```
[the remark this answers is not in the document]
Comment (philswatton) [reply src/modsim2/attack/adversarial_images.py:162]: fixed

File: src/modsim2/attack/adversarial_images.py [hunk 4]
```diff
@@ -192,12 +204,27 @@ def generate_adversarial_images(
         _, clipped_advs, success = attack(fmodel, images, labels, epsilons=epsilons)
 
     # Apply image selection based on attack choice
-    advs_images, advs_success = select_best_attack(
+    advs_images, _ = select_best_attack(
         images=clipped_advs, success=success, epsilons=epsilons
     )
 
+    # Get the model vulnerability metrics
+    vuln = compute_transfer_attack(
```
[the remark this answers is not in the document]
Comment (philswatton) [reply src/modsim2/attack/adversarial_images.py]: Have renamed to `compute_attack_metrics` (and rename some files accordingly)

**Comment by philswatton** [comment]
Have done the latest round of corrections and also fixed the issue with Boundary Attack vulnerability always being recorded as 0. Have run the drop-only group already and can confirm it is no longer 0 vulnerability

**Pull request merged**